tests/ci/test_digest.py

```diff
@@ -99,9 +71,9 @@ def test_digest_path(self):
         # ├── dir3 -> dir2
         # └── symlink-12 -> 12
         hash_expected = md5()
-        hash_expected.update(_12 * 3 + (_13 + _12) * 2)
+        hash_expected.update(_12 * 3 + _13 + (_12 + _13) + _12)
```

Comment:
If the value is changed, then it's broken

Reply:
this is equal expression, i just changed while debugging

Reply:
you are right. I am bad at arithmetic.

However, to make tests more correct, and to catch unnecessary recursion, let's add another subdirectory in one of `dir1` or `dir2`. Better in `dir1`.

Reply:
The mentioned additional test against recursion is done in https://github.com/ClickHouse/ClickHouse/pull/56785/files